### PR TITLE
[GEOS-9253]: Updating Sphinx DOC building instructions

### PR DESCRIPTION
**Pull request by dromagnoli**
## Description

Updating the sphinx installation instruction and needed command to build the GeoServer documentation

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)

The following are required only for core and extension modules, while they are welcomed, but not required, for community modules:
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates

Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.

### CODE REVIEW

**Comment by dromagnoli** [comment]
@jodygarnett can you review this PR and check if the updates I made for Windows and Linux can also be applied to the MacOSX Section? I don't have a MacOSX to test if "pip install sphinx" works fine.

File: doc/en/docguide/source/install.rst [hunk 1]
```diff
@@ -60,8 +56,7 @@ Ubuntu
 Mac OS X
 --------
 
-Installing Sphinx on Mac OS X is nearly identical to installing Sphinx on a 
-Linux system. 
+Installing Sphinx on Mac OS X is nearly identical to installing Sphinx on a Linux system.
```
Comment (tbarsballe) [inline doc/en/docguide/source/install.rst]: I don't think this line is correct anymore, or at least it won't be if you are just using apt to install sphinx on ubuntu

File: doc/en/docguide/source/install.rst [hunk 2]
```diff
@@ -39,19 +35,19 @@ Ubuntu
 
 #. Open a terminal and type the following command::
   
-      sudo apt-get install python-dev build-essential python-setuptools 
+      sudo apt-get install python-dev build-essential
```
[the remark this answers is not in the document]
Comment (dromagnoli) [reply doc/en/docguide/source/install.rst]: Doh! Just tried that on my Ubuntu 16.04 and it messed up all my sphinx installation :| 
throwing an ImportError: cannot import name main.
After a few workaround found on the web, it wasn't finding Python3 so I have re-installed python 3 and it's still not working, so I tried pip install again and it isn't finding pip anymore :)

I'm going to take a look on that again tomorrow but at the moment I'm against that command :)

File: doc/en/docguide/source/install.rst [hunk 3]
```diff
@@ -60,8 +56,7 @@ Ubuntu
 Mac OS X
 --------
 
-Installing Sphinx on Mac OS X is nearly identical to installing Sphinx on a 
-Linux system. 
+Installing Sphinx on Mac OS X is nearly identical to installing Sphinx on a Linux system. 
 
 Easy install
```
[the remark this answers is not in the document]
Comment (dromagnoli) [reply doc/en/docguide/source/install.rst]: Thx. I will remove the manual section,then

File: doc/en/docguide/source/install.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (tbarsballe) [reply doc/en/docguide/source/install.rst]: Fair enough; I was mostly going off of what the Sphinx docs recommended. I think thats worked for me before on a fresh Ubuntu install (VM), but I have no idea how it interacts with an existing Sphinx/Python install.

**Comment by dromagnoli** [comment]
@tbarsballe I have updated the PR with the changes based on your feedbacks. Thanks for that.
If you think it's ok, I will merge it

**Comment by jodygarnett** [comment]
I tend to use home-brew to install python and sphinx-build on macOS.

**Comment by dromagnoli** [comment]
Hi @jodygarnett , do you have any instruction we want to add into a new "home-brew" section for MacOSX  or are you fine with the current state of the PR?

**Comment by jodygarnett** [comment]
Here are some instructions for macOS:

```rst

Users of `homebrew <https://brew.sh>`__ package manager can install with::

    brew install python3

Then use ``pip3`` to install :command:`sphinx`::

    pip3 install sphinx

Confirm availability with::
   
   sphinx-build --version

::

   sphinx-build 1.8.4
```

**Comment by dromagnoli** [comment]
@jodygarnett thanks for the updated instructions. I have updated the MacOSX paragraph with your content.
Let me know if looks good.

**Pull request merged**